in_toto/verifylib.py

```diff
@@ -1112,6 +1120,11 @@ def verify_item_rules(source_name, source_type, rules, links):
         " one of 'materials' or 'products.'\n"
         "Got:\n\t'{}'".format(source_type))
 
+  # Initialize rule traceback dictionary
+  RULE_TRACE["name"] = source_name
+  RULE_TRACE["type"] = source_type
+  RULE_TRACE["trace"] = list()
+  results_dict = {}
```

Comment:
You must re-initialize `results_dict` inside the loop for each iteration, otherwise you'll update the same dict over and over again and all elements in your list (references) will point to the same dict:

```python
# What you are doing
foo = {}
baz = []
for i in range(5):
  foo["bar"] = i
  baz.append(foo)
print(baz)
# Output: [{'bar': 4}, {'bar': 4}, {'bar': 4}, {'bar': 4}, {'bar': 4}]
```
vs.

```python
# What you should do
baz = []
for i in range(5):
  foo = {}
  foo["bar"] = i
  baz.append(foo)
print(baz)
# Output: [{'bar': 0}, {'bar': 1}, {'bar': 2}, {'bar': 3}, {'bar': 4}]
```